### PR TITLE
Fix for error in deleting sandboxes

### PR DESCRIPTION
**Pull request by jontours**
Checks if space services are either bound or contain service keys, if so deletes them and then the service.

### CODE REVIEW

File: sandbox/sandbox.go [hunk 1]
```diff
@@ -53,8 +53,35 @@ func PurgeSpace(client *cfclient.Client, space cfclient.Space) error {
...
 		query := url.Values(map[string][]string{"q": []string{fmt.Sprintf("space_guid:%s", space.Guid)}})
 		apps, err := client.ListAppsByQuery(query)
+		services, err1 := client.ListServiceInstancesByQuery(query)
 		if err != nil {
 			return err
+		}
+		if err1 != nil {
+			return err1
+		}
+		for _, service := range services {
```
Comment (jmcarp) [inline sandbox/sandbox.go]: If we delete apps before services, we shouldn't have to check for app bindings and unbind them.

**Comment by jmcarp** [comment]
What problems does this fix?

**Comment by jmcarp** [comment]
Also, did you get a chance to run the tests for this? I'm getting syntax errors on my end.

**Comment by jontours** [comment]
Some of the spaces can't be deleted as the service keys (if they exist) need to be deleted first.

**Comment by jmcarp** [comment]
So deleting a space doesn't delete service keys within the space?

**Comment by jontours** [comment]
The behavior we saw was with a service binding: 
```
Service broker failed to delete service binding for instance my-service-account: Service broker
```

But now I'm starting to think this PR wouldn't solve this.

**Comment by jontours** [comment]
But that error did happen because of the existence of a service key, as we verified.

**Comment by jontours** [comment]
So to answer your question, no, deleting a space fails to delete a service key if it exists.

**Comment by jmcarp** [comment]
I just tried this out manually, and I don't think that's right--I can create a new space, add a service, add a binding, and delete the space without an error. Like I said on slack, I'm guessing this is specific to s3 service instances failing to delete when the buckets aren't empty.

**Comment by jontours** [comment]
Yes, I think you are write about bindings.  However I think this problem does exists _if_ a service exists within that space that has a service key and you try to delete the space.  @rogeruiz and I saw verified this.

**Comment by jmcarp** [comment]
That's exactly what I just tried, and it worked. Maybe you can show me what you did or double-check?

**Comment by jontours** [comment]
Yeah, you're right, I did just verify.  I think the behavior we saw may have thrown us off track a bit.

**Comment by rogeruiz** [comment]
Thanks @jontours. You have some time to pair on this Monday?

**Comment by jontours** [comment]
Yup! @rogeruiz

**Comment by rogeruiz** [comment]
Closing this as we've solved this issue in the broker rather than the purger. cloudfoundry-community/s3-broker#33.

**Pull request closed without merging**